app/models/user.rb

```diff
@@ -2,4 +2,10 @@ class User < ApplicationRecord
   has_secure_password
 
   validates :email, :user_name, presence: true, uniqueness: true
+  validates_format_of :email, with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i
```

Comment:
Suggest you to place `/\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i` into constant `EMAIL_REGEX`, but it's on your consideration.